File: app-java/src/main/java/com/tritiumstudios/phasmophobiaevidencetool/android/activities/mainmenus/MainMenuFirebaseFragment.java
package com.tritiumstudios.phasmophobiaevidencetool.android.activities.mainmenus;

import androidx.lifecycle.ViewModelProvider;

import com.tritiumstudios.phasmophobiaevidencetool.android.activities.pet.FirebaseFragment;
import com.tritiumstudios.phasmophobiaevidencetool.android.data.viewmodels.MainMenuViewModel;

/**
 * TitleScreenFragment class
 *
 * @author TritiumGamingStudios
 */
public abstract class MainMenuFirebaseFragment extends FirebaseFragment {

    protected MainMenuViewModel mainMenuViewModel;

    public MainMenuFirebaseFragment() {
        super();
    }

    public MainMenuFirebaseFragment(int layout) {
        super(layout);
    }

    protected void initViewModels() {
        super.initGlobalPreferencesViewModel();
    }

    protected void initMainMenuViewModel() {
        if (mainMenuViewModel == null) {
            try {
                mainMenuViewModel =
                        new ViewModelProvider(requireActivity()).get(MainMenuViewModel.class);
            } catch (IllegalStateException e) {
                e.printStackTrace();
            }
        }
    }

    @Override
    protected void backPressedHandler() {
        super.backPressedHandler();
    }

    @Override
    protected void saveStates() {}

}

File: app-view/src/main/java/com/tritiumgaming/phasmophobiaevidencepicker/presentation/ui/activities/mainmenus/MainMenuFirebaseFragment.kt
package com.tritiumgaming.phasmophobiaevidencepicker.presentation.ui.activities.mainmenus

import androidx.fragment.app.activityViewModels
import androidx.fragment.app.viewModels
import androidx.lifecycle.ViewModelProvider
import com.tritiumgaming.phasmophobiaevidencepicker.presentation.ui.activities.impl.AccountManagementService
import com.tritiumgaming.phasmophobiaevidencepicker.presentation.ui.activities.pet.FirebaseFragment
import com.tritiumgaming.phasmophobiaevidencepicker.presentation.viewmodel.MainMenuViewModel

abstract class MainMenuFirebaseFragment : FirebaseFragment {

    protected val mainMenuViewModel: MainMenuViewModel by activityViewModels()

    constructor() : super()

    constructor(layout: Int) : super(layout)

    override fun initViewModels() { super.initGlobalPreferencesViewModel() }

    override fun saveStates() {}
}